python/setup.py

```diff
@@ -201,7 +201,7 @@ def _supports_symlinks():
         extras_require={
             'ml': ['numpy>=1.7'],
             'mllib': ['numpy>=1.7'],
-            'sql': ['pandas>=0.19.2']
+            'sql': ['pandas>=0.19.2', 'pyarrow>=0.8.0']
```

Comment:
If no pyarrow is installed, will setup force users to install it?

Reply:
Nope, `extras_require` does not do anything in normal cases but they can be installed together with a dev option via pip IIRC.